vars/dockerExecute.groovy

```diff
@@ -1,53 +1,85 @@
 import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.JenkinsUtils
+import groovy.transform.Field
 
-def call(Map parameters = [:], body) {
+@Field def STEP_NAME = 'dockerExecute'
+@Field def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
 
-    def STEP_NAME = 'dockerExecute'
-    def PLUGIN_ID_DOCKER_WORKFLOW = 'docker-workflow'
+@Field Set GENERAL_CONFIG_KEYS = ['jenkinsKubernetes']
 
-    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters){
-        def dockerImage = parameters.dockerImage ?: ''
-        Map dockerEnvVars = parameters.dockerEnvVars ?: [:]
-        def dockerOptions = parameters.dockerOptions ?: ''
-        Map dockerVolumeBind = parameters.dockerVolumeBind ?: [:]
+@Field Set PARAMETER_KEYS = ['dockerImage',
+                             'dockerOptions',
+                             'dockerWorkspace',
+                             'dockerEnvVars',
+                             'dockerVolumeBind']
+@Field Set STEP_CONFIG_KEYS = PARAMETER_KEYS
 
-        if(dockerImage) {
+void call(Map parameters = [:], body) {
+    handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters) {
+        final script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
 
-            if (! isPluginActive(PLUGIN_ID_DOCKER_WORKFLOW) ) {
-                echo "[WARNING][${STEP_NAME}] Docker not supported. Plugin '${PLUGIN_ID_DOCKER_WORKFLOW}' is not installed or not active. Configured docker image '${dockerImage}' will not be used."
-                dockerImage = null
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, GENERAL_CONFIG_KEYS)
+            .mixinStageConfig(script.commonPipelineEnvironment, parameters.stageName?:env.STAGE_NAME, STEP_CONFIG_KEYS)
+            .mixinStepConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
+            .mixin(parameters, PARAMETER_KEYS)
+            .withMandatoryProperty('dockerImage')
+            .use()
+
+        if (isKubernetes() && config.dockerImage) {
+            if (env.POD_NAME && isContainerDefined(config)) {
+                container(getContainerDefined(config)) {
+                    echo "Executing inside a Kubernetes Container"
+                    body()
+                    sh "chown -R 1000:1000 ."
```

Comment:
Is ```1000:1000```always right, or may that depend on the customer setup of the images. In case ```1000:1000``` is appropriate always it should be explained here, maybe. Otherwise it should be configurable.

Reply:
As of now. We are proceeding with an assumption that the docker container should have the user 1000. This will be explained here and in docs.